File: simple-datawarehouse-backend/src/main/java/pl/edu/agh/simpledatawarehouse/model/dto/PivotTableQuery.java
package pl.edu.agh.simpledatawarehouse.model.dto;

import pl.edu.agh.simpledatawarehouse.model.query.Query;

import java.util.List;

public record PivotTableQuery(
        Query query,
        List<String> rowLabels,
        List<String> columnLabels,
        List<String> valueLabels) {

    public boolean isPivoted() {
        return !rowLabels.isEmpty();
    }

}

File: simple-datawarehouse-frontend/src/app/_models/pivot/pivot-table-query.ts
import {Query} from "@app/_models/query/query";

export class PivotTableQuery {
  constructor(
    public query: Query = {} as Query,
    public rowLabels: string[] = Array.of(),
    public columnLabels: string[] = Array.of(),
    public valueLabels: string[] = Array.of()
  ) {
  }

  hasRequiredSearchData(): boolean {
    if (this.rowLabels.length === 0
      && this.columnLabels.length === 0
      && this.valueLabels.length === 0) {
      return false;
    }
    return true;
  }
}
